pkg/controller/logstorage/dashboards/dashboards_controller_test.go

```diff
@@ -0,0 +1,511 @@
+			mockStatus.On("SetDegraded", mock.Anything, mock.Anything, mock.Anything, mock.Anything)
+			mockStatus.On("ClearDegraded")
+
+			// Create a CA secret for the test, and create its KeyPair.
+			opts := []certificatemanager.Option{
+				certificatemanager.AllowCACreation(),
+				certificatemanager.WithTenant(tenant),
+			}
+			cm, err := certificatemanager.Create(cli, &install.Spec, dns.DefaultClusterDomain, tenantNS, opts...)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(cli.Create(ctx, cm.KeyPair().Secret(tenantNS))).ShouldNot(HaveOccurred())
+
+			// Create secrets needed for successful installation.
+			bundle := cm.CreateTrustedBundle()
+			Expect(cli.Create(ctx, bundle.ConfigMap(tenantNS))).ShouldNot(HaveOccurred())
+
+			// Create the ES user secret. Generally this is created by either es-kube-controllers or the user controller in this operator.
+			userSecret := &corev1.Secret{}
+			userSecret.Name = dashboards.ElasticCredentialsSecret
+			userSecret.Namespace = tenantNS
+			userSecret.Data = map[string][]byte{"username": []byte("test-username"), "password": []byte("test-password")}
+			Expect(cli.Create(ctx, userSecret)).ShouldNot(HaveOccurred())
+
+			// Create the reconciler for the test.
+			r, err = NewDashboardsControllerWithShims(cli, scheme, mockStatus, operatorv1.ProviderNone, dns.DefaultClusterDomain, true, false)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should wait for the tenant CA to be provisioned", func() {
+			// Delete the CA secret for this test.
+			caSecret := &corev1.Secret{}
+			caSecret.Name = certificatemanagement.TenantCASecretName
+			caSecret.Namespace = tenantNS
+			Expect(cli.Delete(ctx, caSecret)).ShouldNot(HaveOccurred())
+
+			// Run the reconciler.
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default", Namespace: tenantNS}})
+			Expect(err).Should(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("CA secret"))
+		})
+
+		It("should not reconcile any resources if no Namespace was given", func() {
+			// Run the reconciler, passing in a Request with no Namespace. It should return successfully.
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Check that nothing was installed on the cluster.
+			dashboardJob := batchv1.Job{
+				TypeMeta: metav1.TypeMeta{Kind: "Job", APIVersion: "batch/v1"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dashboards.Name,
+					Namespace: render.ElasticsearchNamespace,
+				},
+			}
+			err = cli.Get(ctx, types.NamespacedName{Name: dashboardJob.Name, Namespace: dashboardJob.Namespace}, &dashboardJob)
+			Expect(err).Should(HaveOccurred())
+			Expect(errors.IsNotFound(err)).Should(BeTrue())
+
+			// Check that OnCRFound was not called.
+			mockStatus.AssertNotCalled(GinkgoT(), "OnCRFound")
+		})
+
+		It("should reconcile resources for a tenant", func() {
```

Comment:
I don't think this test is correct - it seems to claim that this works for a tenant, but actually this controller doesn't reconcile for multi-tenant envs.

Reply:
We are disabling multi-tenancy in Add() and testing out Reconcile(). Seems like there is mismatch between this two.

Reply:
We disable multi-tenancy in Add() and test calling Reconcile(). Seems like the is a mismatch between these two.